Userland/Libraries/LibAudio/GenericTypes.cpp

```diff
@@ -27,7 +27,9 @@ Optional<SeekPoint const&> SeekTable::seek_point_before(u64 sample_index) const
         return {};
     size_t nearby_seek_point_index = 0;
     AK::binary_search(m_seek_points, sample_index, &nearby_seek_point_index, [](auto const& sample_index, auto const& seekpoint_candidate) {
-        return static_cast<i64>(sample_index) - static_cast<i64>(seekpoint_candidate.sample_index);
+        // Subtraction with i64 cast may cause overflow.
+        return sample_index > seekpoint_candidate.sample_index ? 1
+                                                               : (sample_index == seekpoint_candidate.sample_index ? 0 : -1);
```

Comment:
Instead of this weirdly formatted nested ternary, I'd prefer:

```cpp
if (sample_index > seekpoint_candidate.sample_index)
    return 1;
return (sample_index == seekpoint_candidate.sample_index ? 0 : -1;
```

Or even

```cpp
if (sample_index > seekpoint_candidate.sample_index)
    return 1;
if (sample_index == seekpoint_candidate.sample_index)
    return 0;
return -1;
```

Reply:
Defense: clang-format made up that indentation :)